### PR TITLE
boards: move board specific configurations

### PR DESCRIPTION
**Pull request by uLipe**
Te idea is to keep consistent with other zephyr modules and separate board specific details into their own overlays.

Also add the example l475 IoT discovery board as an example.

No changes in the environment or build.

### CODE REVIEW

**Comment by uLipe** [comment]
@pablogs9 @Acuadros95 PTAL.

**Comment by pablogs9** [comment]
Running CI, thanks a lot @uLipe

**Comment by pablogs9** [comment]
@mergify backport foxy galactic main

**Comment by pablogs9** [comment]
It seems to break the CI @uLipe : https://github.com/micro-ROS/micro_ros_zephyr_module/runs/7534484810?check_suite_focus=true#step:4:5888

**Comment by uLipe** [comment]
@pablogs9 my bad, during rebase I dropped a piece of the dts I will change this to draft while I fix it

**Comment by pablogs9** [comment]
Thanks. Maybe we should investigate the working status of this module in 2.7 @Acuadros95

**Comment by uLipe** [comment]
@pablogs9 PTAL again, for now let's keep compatible with 2.6.

Actually I have this running on Zephyr 3.1 but using the overlay + kconfig updates that broke the CI + this PR https://github.com/ros2/rcutils/pull/368

**Comment by pablogs9** [comment]
Could you take a look at this https://github.com/micro-ROS/micro_ros_zephyr_module/pull/64 @uLipe. We would like to have this working with v3.1.0 as you have.

**Comment by uLipe** [comment]
@pablogs9 , @Acuadros95 , I just took the opportunity to modify the zephyr transport serial port to use the DT macros, label property in the devices are being removed from Zephyr and will not work in future kernel releases.

Please take another review and ask me any questions you might to have.

**Comment by pablogs9** [comment]
I'll take a look on Monday when I have some hardware in the office

File: modules/libmicroros/microros_transports/serial-usb/microros_transports.c [hunk 1]
```diff
@@ -53,12 +53,13 @@ static void uart_fifo_callback(struct device *dev){
...
-
     int ret;
     uint32_t baudrate, dtr = 0U;
 
-    params->uart_dev = device_get_binding("CDC_ACM_0");
+    /* for serial-usb transport we just override the device pointer
+     * with USB to use the same interface
+     */
+    transport->args = device_get_binding("CDC_ACM_0");
     if (!params->uart_dev) {
```
Comment (pablogs9) [inline modules/libmicroros/microros_transports/serial-usb/microros_transports.c:63]: `params` undefined here

File: modules/libmicroros/microros_transports/serial-usb/microros_transports.c [hunk 2]
```diff
@@ -76,7 +77,7 @@ bool zephyr_transport_open(struct uxrCustomTransport * transport){
     printk("Waiting for agent connection\n");
 
     while (true) {
-        uart_line_ctrl_get(params->uart_dev, UART_LINE_CTRL_DTR, &dtr);
+        uart_line_ctrl_get(uart_dev, UART_LINE_CTRL_DTR, &dtr);
```
Comment (pablogs9) [inline modules/libmicroros/microros_transports/serial-usb/microros_transports.c:80]: Where is `uart_dev` defined?

**Review by pablogs9 — CHANGES_REQUESTED** [review]

**Comment by pablogs9** [comment]
Hello @uLipe:
- I have been able to test this with serial transport in `disco_l475_iot1`
- **Not** able to build the serial-usb example
- Should we drop the support for v2.4.0 if we merge this PR?

**Comment by uLipe** [comment]
Hi @pablogs9 any updates regarding your tests?

About the Zephyr version I think it is safe to drop the 2.4 support, since most of V2 users would go to version 2.6 or 2.7, the last one is a LTS version.

There is no Zephyr-API breaking changes across 2.4 to 2.7, the same is not true for the build system for example, depending on what changed in a particular subsystem.

**Comment by pablogs9** [comment]
Hello @uLipe, I'll check this on Monday in the office, because I have no boards right now.

Meanwhile, I would like to drop 2.4 and add 2.7 in this PR: https://github.com/micro-ROS/micro_ros_zephyr_module/pull/64 is it possible for you to take a look there?

**Comment by pablogs9** [comment]
Hello @uLipe, I have been able to build the USB example but I have the following error:

```
*** Booting Zephyr OS build v2.6.0-rc1  ***
Failed to enable USB
```

Any idea? I'm working in the disco_l475_iot1

**Comment by uLipe** [comment]
What is the ERRNO @pablogs9 ?  Have you tried in a different board like that?

**Comment by pablogs9** [comment]
That's our reference board for Zephyr, not sure if I have other. `errno` is 0 (?!?!)

```c
    ret = usb_enable(NULL);
    if (ret != 0) {
        printk("Failed to enable USB %d\n",errno);
        return false;
    }
```

result in 

```
*** Booting Zephyr OS build v2.6.0-rc1  ***
Failed to enable USB 0
```

**Comment by uLipe** [comment]
actually ret comes with the error code translated into errno code, could you print ret instead?

My board currently opens and attches to the usb correctly.

**Comment by pablogs9** [comment]
```c
    ret = usb_enable(NULL);
    if (ret != 0) {
        printk("Failed to enable USB %d\n", ret);
        return false;
    }
```

results in 

```
*** Booting Zephyr OS build v2.6.0-rc1  ***
Failed to enable USB -5
```

Any idea?

**Comment by pablogs9** [comment]
I have been testing `samples/subsys/usb/cdc_acm` with my board and I have:

```
*** Booting Zephyr OS build zephyr-v3.1.0  ***
[00:00:00.034,000] <err> usb_dc_stm32: PCD_Init failed, 1
[00:00:00.034,000] <err> cdc_acm_echo: Failed to enable USB
```

This does not seems like a micro-ROS transport problem.

**Comment by pablogs9** [comment]
@uLipe I have updated the support here: https://github.com/micro-ROS/micro_ros_zephyr_module/pull/64

Please rebase this PR and we can merge since my USB-CDC problems are not related

BTW I have backported https://github.com/ros2/rcutils/pull/368 to `foxy`, `humble` and `galactic` in the micro-ROS fork.

**Comment by uLipe** [comment]
@pablogs9 btw the error you re receiving is a timeout that is coming from USB endopoint, code 5 is -EIO, which is generated by the CDC ACM class when starting endpoint sending and no notification about the write is read back, are the connections, cable ot host computer okay? Do you see the enumeration on your host?

**Comment by uLipe** [comment]
@pablogs9 rebase, please check one more time.

**Review by pablogs9 — CHANGES_REQUESTED** [review]

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,9 +2,8 @@
 Changelog for package micro_ros_zephyr
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-3.0.0 (2022-05-25)
+4.0.0 (2022-05-25)
 ------------------
-* Initial humble release (`#59 <https://github.com/micro-ROS/micro_ros_zephyr_module/issues/59>`_)
 * Update banner (`#56 <https://github.com/micro-ROS/micro_ros_zephyr_module/issues/56>`_)
```
[the remark this answers is not in the document]
Comment (uLipe) [reply CHANGELOG.rst:7]: Probably messed up the things during the rebase, will change that

**Comment by uLipe** [comment]
@pablogs9 fixed the changelog, please do a check again.

File: CHANGELOG.rst [hunk 4]
```diff
@@ -2,7 +2,7 @@
 Changelog for package micro_ros_zephyr
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-3.0.0 (2022-05-25)
+4.0.0 (2022-05-25)
```
Comment (pablogs9) [inline CHANGELOG.rst:5]:
```suggestion
3.0.0 (2022-05-25)
```

**Review by pablogs9 — APPROVED** [review]
LGTM, @Acuadros95 merge if you are ok

**Comment by Acuadros95** [comment]
@uLipe Could you take a look at the CI? [detail](https://github.com/micro-ROS/micro_ros_zephyr_module/runs/7786824074?check_suite_focus=true#step:4:6478)

```bash
/__w/micro_ros_zephyr_module/micro_ros_zephyr_module/micro_ros_zephyr_module/boards/disco_l475_iot1.conf:1: warning: attempt to assign the value 'y' to the undefined symbol USB
...

error: Aborting due to Kconfig warnings
```

**Comment by uLipe** [comment]
@pablogs9 sorry for the delay, since we dropped the version 2.6 the CONFIG_USB symbol was also deprecated, I will remove it and update the PR

**Comment by uLipe** [comment]
@pablogs9 @Acuadros95 I added the Zephyr UDC0 nodelabel, which replaces the CONFIG_USB from version 2.7 and above, please approve the workflow when you have time.

File: CHANGELOG.rst (hunk 4 above)
Comment (pablogs9) [inline CHANGELOG.rst:5]:
```suggestion
3.0.0 (2022-05-25)
```

**Review by pablogs9 — APPROVED** [review]

File: CHANGELOG.rst (hunk 4 above)
Comment (pablogs9) [inline CHANGELOG.rst:5]:
```suggestion
3.0.0 (2022-05-25)
```

**Comment by pablogs9** [comment]
@Acuadros95 PTAL and merger if you are ok

**Review by Acuadros95 — APPROVED** [review]
LGTM

**Pull request merged**